compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -1010,9 +1010,14 @@ class RefChecks extends MiniPhase { thisPhase =>
   }
 
   override def transformNew(tree: New)(implicit ctx: Context) = {
-    val sym = tree.tpe.typeSymbol
+    val tpe = tree.tpe
+    val sym = tpe.typeSymbol
     checkUndesiredProperties(sym, tree.pos)
     currentLevel.enterReference(sym, tree.pos)
+    tpe.dealias.foreachPart {
```

Comment:
Is `dealias` required?

Reply:
Without it the second test still fails. If you would dealias a bit sooner I think `def foo = { type T = Foo; val a = new T; class Foo }` would also emit an error, like in scalac.